tools/enrollment_preprocessor.py

```diff
@@ -0,0 +1,160 @@
+
+
+class UserCells(NamedTuple):
+    title: Cell | None
+    last_name: Cell
+    first_name: Cell
+    email: Cell
+
+    def _clean(self) -> Iterator[str]:
+        for field in iter(self):
+            if not field or not field.value:
+                yield ""
+                continue
+            field.value = str(field.value).strip()
+            yield field.value
+
+    def clean_user(self):
+        return User(*self._clean())
+
+
+def user_from_row(row: tuple[Cell, ...]):
+    return [
+        UserCells(None, *row[:3]).clean_user(),
+        UserCells(*row[7:]).clean_user(),
+    ]
+
+
+def make_bold(text: str) -> str:
+    return f"\033[1m{text}\033[0m"
+
+
+def csv_users_by_email(user_data: TextIO) -> dict[str, User]:
+    users = {}
+    reader = csv.reader(user_data, delimiter=";", lineterminator="\n")
+    next(reader)  # skip header
+    for row in reader:
+        user = User(*row)
+        users[user.email] = user
+    return users
+
+
+def get_user_decisions(database_users: dict[str, User], workbook: Workbook) -> dict[str, User]:
+    conflicts_by_field: dict[str, list[User]] = defaultdict(list)
+    for sheet in workbook.worksheets:  # parse enrollment data and group conflicts
+        for imported in chain.from_iterable(map(user_from_row, sheet.iter_rows(min_row=2, min_col=2))):
+            if not imported.email:
+                continue
+            existing = database_users.setdefault(imported.email, imported)
+            if imported == existing:
+                continue
+            for field in ["title", "last_name", "first_name"]:
+                field_value = getattr(imported, field)
+                if field_value is not None and getattr(existing, field) != getattr(imported, field):
+                    conflicts_by_field[field].append(imported)
+    # ask user for decision
+    for field, conflicts in conflicts_by_field.items():
+        print(field.capitalize())
+        print("---------")
+        for imported in conflicts:
+            existing = database_users[imported.email]
+
+            if getattr(existing, field) == getattr(imported, field):
+                continue
+
+            print(f"existing: '{make_bold(getattr(existing, field))}' ({existing.full_name()})")
+            print(f"imported: '{make_bold(getattr(imported, field))}' ({imported.full_name()})")
+
+            decision = ""
+            while decision not in ("e", "i"):
+                decision = input("Which one should be used? (e/i):\n")
+            choice = existing if decision == "e" else imported
+
+            setattr(database_users[choice.email], field, getattr(choice, field))
+        print()
+    return database_users
+
+
+def update_cells(cells: UserCells, user: User) -> bool:
+    changed = False
+    for cell, field_value in zip(cells, [user.title, user.last_name, user.first_name, user.email], strict=True):
+        if cell and cell.value != field_value:
```

Comment:
Could we do
```suggestion
        if cell is not None and cell.value != field_value:
```
to make our intention more explicit